test/mmio/mmio.c

```diff
@@ -0,0 +1,510 @@
+/*
+*   Matrix Market I/O library for ANSI C
+*
+*   See http://math.nist.gov/MatrixMarket for details.
```

Comment:
We should add the license for this.

Reply:
It might be better if this was added in a separate PR. Then this funcitonallity is not tied to this feature and can be backported.

Reply:
I didn't find any mention of license in https://math.nist.gov/MatrixMarket/info.html#Credits other than credits.

MTX file reading is not mixed with sparse-sparse arithmetic changes.

This PR has three different commits.
- Removes deprecated MKL API
- Sparse-Sparse arith
- Support to read MTX files and tests that use MTX files. If we separate the tests, then it can be backported.

Reply:
Separated the matrixmarket changes into separate commit and sparse-arith tests based on them into different commit.